compressors/huffman_coder.py

```diff
@@ -0,0 +1,145 @@
+from dataclasses import dataclass
+from typing import Any
+from core.data_stream import BitsDataStream, DataStream
+from core.prob_dist import ProbabilityDist
+from core.data_compressor import DataCompressor
+from core.data_transformer import (
+    BitstringToBitsTransformer,
+    CascadeTransformer,
+    BitsParserTransformer,
+    LookupTableTransformer,
+)
+
+
+@dataclass
+class HuffmanNode:
```

Comment:
@shubhamchandak94: Let me know if this makes more sense.

Reply:
Thanks, looks good